concent_api/core/models.py

```diff
@@ -28,6 +33,43 @@
 from .constants             import MESSAGE_TASK_ID_MAX_LENGTH
 
 
+class SubtaskWithTimingColumnsManager(Manager):
+    """Creates maximum_download time, subtask_verification_time and download_deadline columns
+    maximum_download_time = DOWNLOAD_LEADIN_TIME + ceil((result_package_size / MINIMUM_UPLOAD_RATE << 10))
+    subtask_verification_time = (4 * CONCENT_MESSAGING_TIME) + (3 * maximum_download_time) + (0.5 * (computation_deadline - task_to_compute_timestamp))
+    download_deadline = computation_deadline + maximum_download_time
+    """
+
+    def with_maximum_download_time(self):  # pylint: disable=no-self-use
+        bytes_per_sec = settings.MINIMUM_UPLOAD_RATE << 10
+        download_time = Func(F('result_package_size') / float(bytes_per_sec), function='CEIL')
+        return Value(settings.DOWNLOAD_LEADIN_TIME) + download_time
+
+    def with_subtask_verification_time(self):  # pylint: disable=no-self-use
+        computation_deadline = F('computation_deadline')
+        task_to_compute_timestamp = Func(Value('epoch'), F('task_to_compute__timestamp'), function='DATE_PART')
+        subtask_timeout = Func(Value('epoch'), computation_deadline, function='DATE_PART') - task_to_compute_timestamp
+        subtask_verification_time = (4 * settings.CONCENT_MESSAGING_TIME) + (3 * F('maximum_download_time')) + (
+                0.5 * subtask_timeout)
```

Comment:
Please fix your indentation and wrapping. It should look like this according to our coding style if you really want to break it where you did:

``` python
subtask_verification_time = (4 * settings.CONCENT_MESSAGING_TIME) + (3 * F('maximum_download_time')) + (
    0.5 * subtask_timeout
)
```

But this would be more readable:
``` python
subtask_verification_time = (
    4 * settings.CONCENT_MESSAGING_TIME +
    3 * F('maximum_download_time') +
    0.5 * subtask_timeout
)
```